brjs-sdk/sdk/libs/javascript/br/src/br/AliasRegistryClass.js

```diff
@@ -128,13 +127,24 @@ AliasRegistryClass.prototype.isAlias = function isAlias(aliasName) {
 * @type boolean
 */
 AliasRegistryClass.prototype.isAliasAssigned = function isAliasAssigned(aliasName) {
-	return this.isAlias(aliasName) && this._aliasData[aliasName]["class"] !== undefined;
+	var aliasProviderFunction = typeof this._aliasData[aliasName] === 'function';
+
+	return this.isAlias(aliasName) && (aliasProviderFunction || this._aliasData[aliasName]["class"] !== undefined);
+};
+
+AliasRegistryClass.prototype.registerAliasProvider = function(aliasName, aliasProvider) {
+	this._aliasData[aliasName] = aliasProvider;
 };
 
 /**
  * @private
  */
 AliasRegistryClass.prototype._getClassRef = function(aliasName) {
+	// Do you access the alias via a provider function.
+	if (typeof this._aliasData[aliasName] === 'function') {
```

Comment:
With this change, when a call is made to `AliasRegistry#getClass(alias)` from `ServiceRegistry`, we will no longer verify that the class referenced by the alias adheres to any implemented interfaces (as there is no longer an interface property to verify against in alias data).